game-core/src/main/java/games/strategy/triplea/delegate/battle/subs/Subs.java

```diff
@@ -0,0 +1,91 @@
+package games.strategy.triplea.delegate.battle.subs;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.MustFightBattle.ReturnFire;
+import java.util.Collection;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class Subs {
+
+  public enum FireOrder {
+    DEF_BEFORE_ATT,
+    DEF_BEFORE_REGULAR,
+    DEF_WITH_REGULAR
+  }
+
+  public FireOrder getFireOrder(
+      final Collection<Unit> attackingUnits,
+      final Collection<Unit> defendingUnits,
+      final GameData gameData) {
+    if (defenderSubsFireFirst(attackingUnits, defendingUnits, gameData)) {
+      return FireOrder.DEF_BEFORE_ATT;
+    }
+    final boolean defendingSubsFireWithAllDefenders =
+        !Properties.getWW2V2(gameData)
+            && returnFireAgainstDefendingSubs(attackingUnits, defendingUnits, gameData)
+                == ReturnFire.ALL;
+    if (defendingSubsSneakAttack(gameData) && !defendingSubsFireWithAllDefenders) {
+      return FireOrder.DEF_BEFORE_REGULAR;
+    }
+    return FireOrder.DEF_WITH_REGULAR;
+  }
+
+  public boolean defenderSubsFireFirst(
+      final Collection<Unit> attackingUnits,
+      final Collection<Unit> defendingUnits,
+      final GameData gameData) {
```

Comment:
Would it make sense to make GameData a ctor param so all these methods don't have to take it as a param?

Reply:
I had left Subs as a utility class so all of the functions are static.  I could instantiate Subs once during MustFightBattle's ctor with gameData and then use the instance each time.  Would that be ok?

Reply:
Ah, I hadn't seen that before. It's very confusing to not have the static keyword on a method and have something else cause it to be static. Is there a consensus for the project to use that? If not, I'd prefer the methods be marked as static explicitly.

Reply:
I went through the code and found a few other classes that had the lombok utility class annotation and they still had the static modifier.  So, I'll add the static/final/etc modifiers to be explicit about it.